app/controllers/settings/integration.js

```diff
@@ -44,6 +44,11 @@ export default Controller.extend({
                     });
                 }
 
+                // if the model is deleted, allow the transition to complete
+                if (this.integration.get('isDeleted')) {
+                    return setTimeout(() => transition.retry(), 0);
```

Comment:
Pretty sure this should be like `Ember.run` or something instead?

Reply:
I don't think you need anything here, I would opt for putting a `!isDeleted` conditional in the [`willTransition` hook](https://github.com/TryGhost/Ghost-Admin/blob/190a710c9c9827bba1ba3d6711cbec5e793fcc37/app/routes/settings/integration.js#L49) that decides whether to show the unsaved changes modal

Reply:
Ahhhhh okay :+1: i didn't even know about that- thank you thank you!